src/NHibernate/Action/DelayedPostInsertIdentifier.cs

```diff
@@ -15,20 +16,27 @@ namespace NHibernate.Action
 	[Serializable]
 	public class DelayedPostInsertIdentifier
 	{
-		[ThreadStatic]
-		private static long _Sequence = 0;
+		private static long GlobalSequence = 0;
```

Comment:
Initially I was going to replace `ThreadStatic` with `AsyncLocal`. But decided it's not really necessary here - let's make it global counter across all threads

Reply:
This is what I was going to ask.